src/Catel.Core/Catel.Core.Shared/Data/PropertyDataManager.cs

```diff
@@ -214,6 +242,27 @@ public PropertyData GetPropertyData(Type type, string name)
         }
 
         /// <summary>
+        /// Gets the dynamic properties data.
+        /// </summary>
+        /// <param name="type">The type for which to get the dynamic properties data.</param>
+        /// <returns>The <see cref="IEnumerable{PropertyData}"/> containing dynamic properties data.</returns>
+        public IEnumerable<PropertyData> GetDynamicPropertiesData(Type type)
+        {
+            Argument.IsNotNull("type", type);
+
+            lock (_propertyDataLock)
```

Comment:
This can be an extension method and be implemented like this:

```
var typeInfo = propertyDataManager.GetCatelTypeInfo(type);
return typeInfo .GetCatelProperties().Values.Where(x => x.IsDynamicProperty).ToList();  
 


```